src/caffe/solver.cpp

```diff
@@ -860,6 +860,214 @@ void AdaGradSolver<Dtype>::ComputeUpdateValue(int param_id, Dtype rate) {
 }
 
 template <typename Dtype>
+void AdaDeltaSolver<Dtype>::PreSolve() {
+  // Add the extra history entries for AdaDelta after those from
+  // SGDSolver::PreSolve
+  const vector<shared_ptr<Blob<Dtype> > >& net_params = this->net_->params();
```

Comment:
You should use `const vector<Blob<Dtype>*>& net_params = this->net_->learnable_params();` instead of `const vector<shared_ptr<Blob<Dtype> > >& net_params = this->net_->params();`